tests/codegen/waiter-tests/src/test/kotlin/com/test/WaiterTest.kt

```diff
@@ -107,6 +108,25 @@ class WaiterTest {
         GetEntityResponse { primitives = EntityPrimitives { intEnum = IntEnum.One } },
     )
 
+    // list indexing
+    @Test fun testBooleanListIndexZeroEquals() = successTest(
+        WaitersTestClient::waitUntilBooleanListIndexZeroEquals,
+        GetEntityResponse { lists = EntityLists { booleans = listOf(false) } },
+        GetEntityResponse { lists = EntityLists { booleans = listOf(true) } },
+    )
+
+    @Test fun testBooleanListIndexOneEquals() = successTest(
+        WaitersTestClient::waitUntilBooleanListIndexOneEquals,
+        GetEntityResponse { lists = EntityLists { booleans = listOf(false, false) } },
+        GetEntityResponse { lists = EntityLists { booleans = listOf(true, true) } },
+    )
```

Comment:
**Nit**: This doesn't test that the correct index is being observed since both values in the list are changing at the same time. I recommend testing `[false, false]`, `[true, false]`, and then finally `[false, true]`.